javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java

```diff
@@ -310,7 +311,11 @@ private static String methodParameterDescriptionOrDefault(ClientMethodParameter
         return doc;
     }
 
-    private static String appendOptionalOrRequiredAttribute(boolean isRequired, boolean isRootSchema) {
-        return isRootSchema ? "" : isRequired ? " (Required)" : " (Optional)";
+    private static String appendOptionalOrRequiredAttribute(boolean isRequired, boolean isRequiredForCreate, boolean isRootSchema) {
+        String optionalStr = " (Optional)";
+        if (isRequiredForCreate && !isRequired) {
+            optionalStr = " (Optional, Required on create)";
+        }
+        return isRootSchema ? "" : isRequired ? " (Required)" : optionalStr;
```

Comment:
Not a big fan of ternary operators in ternary operators as it gets hard to read.

Could we just have an early out of `""` at the beginning if it's the root schema? Then just have an if / else if / else block?

```java
if (isRootSchema) {
    return "";
} else if (isRequired) {
    return " (Required)";
} else if (isRequiredForCreate) {
    return " (Optional, Required on create)";
} else {
    return " (Optional)";
}
```

Reply:
Agree. Nested ternary is hard to read (even with parentheses).

Updated.